tests/unit-tests/test_gnn.py

```diff
@@ -2471,6 +2512,448 @@ def check_call_gen_embs(skip_last_self_loop):
+        ninput_feats = ep_data.get_node_feats(input_nodes, nfeat_fields)
+        efeat_fields = dataloader3.edge_feat_fields
+        einput_feats_list = ep_data.get_blocks_edge_feats(blocks, efeat_fields)
+
+        label_field = dataloader3.label_field
+        lbl = ep_data.get_edge_feats({("n0", "r1", "n1"): [0, 1]}, label_field)
+
+        with assert_raises(AssertionError):
+            model3(blocks, batch_graph, ninput_feats, einput_feats_list,
+                   None, lbl, input_nodes)
+
+    # Test case 3: abnormal case, edge model does not use edge feature, but provides edge feature
+    #              for mini-batch forward. Should Trigger an assertion error about initialization
+    #              of RGCN model using `edge_feat_name` argument.
+    edge_feat_name = {('n0', 'r0', 'n1'): 'feat', ('n0', 'r1', 'n1'): 'feat'}
+    model4 = create_rgcn_edge_model(ep_data.g, num_ffn_layers=0, edge_feat_name=None)
+    target_idx = {("n0", "r1", "n1"): [0, 1]}
+    dataloader4 = GSgnnEdgeDataLoader(ep_data,
+                                      target_idx,
+                                      fanout=[-1, -1],
+                                      batch_size=10,
+                                      label_field='label',
+                                      node_feats='feat',
+                                      edge_feats='feat',
+                                      train_task=False,
+                                      remove_target_edge_type=False)
+    for i, (input_nodes, batch_graph, blocks) in enumerate(dataloader4):
+        nfeat_fields = dataloader4.node_feat_fields
+        ninput_feats = ep_data.get_node_feats(input_nodes, nfeat_fields)
+        efeat_fields = dataloader4.edge_feat_fields
+        einput_feats_list = ep_data.get_blocks_edge_feats(blocks, efeat_fields)
+
+        label_field = dataloader4.label_field
+        lbl = ep_data.get_edge_feats({("n0", "r1", "n1"): [0, 1]}, label_field)
+
+        with assert_raises(AssertionError):
+            model4(blocks, batch_graph, ninput_feats, einput_feats_list,
+                   None, lbl, input_nodes)
+
+    th.distributed.destroy_process_group()
+    dgl.distributed.kvstore.close_kvstore()
+
+def test_rgcn_lp_model_forward():
+    # initialize the torch distributed environment
+    th.distributed.init_process_group(backend='gloo',
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+        ep_data = GSgnnData(part_config=part_config,
+                            node_feat_field="feat",
+                            edge_feat_field="feat")
+
+    # Test case 0: normal case, lp model uses edge feature, and provides edge feature
```

Comment:
ditto

Reply:
Add another test case, this will trigger an assert error. In addition, this case has been tested in test_embed.py line https://github.com/awslabs/graphstorm/pull/281.